### PR TITLE
Fix OOB in site picker

### LINKED ISSUE #5971: Crash report 7.4: IndexOutOfBoundsException in SitePickerAdapter.getNumHiddenSelected
```
Fatal Exception: java.lang.IndexOutOfBoundsException: Invalid index 2, size is 2
       at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
       at java.util.ArrayList.get(ArrayList.java:308)
       at org.wordpress.android.ui.main.SitePickerAdapter.getNumHiddenSelected(SitePickerAdapter.java:270)
       at org.wordpress.android.ui.main.SitePickerActivity$ActionModeCallback.onPrepareActionMode(SitePickerActivity.java:511)
       at android.support.v7.app.AppCompatDelegateImplV9$ActionModeCallbackWrapperV9.onPrepareActionMode(AppCompatDelegateImplV9.java:1789)
       at android.support.v7.app.WindowDecorActionBar$ActionModeImpl.invalidate(WindowDecorActionBar.java:1032)
       at org.wordpress.android.ui.main.SitePickerActivity.onSelectedCountChanged(SitePickerActivity.java:440)
       at org.wordpress.android.ui.main.SitePickerAdapter.setItemSelected(SitePickerAdapter.java:308)
       at org.wordpress.android.ui.main.SitePickerAdapter.access$1100(SitePickerAdapter.java:35)
       at org.wordpress.android.ui.main.SitePickerAdapter$2.onLongClick(SitePickerAdapter.java:209)
       at android.view.View.performLongClick(View.java:4895)
       at android.view.View$CheckForLongPress.run(View.java:20242)
       at android.os.Handler.handleCallback(Handler.java:815)
       at android.os.Handler.dispatchMessage(Handler.java:104)
       at android.os.Looper.loop(Looper.java:194)
       at android.app.ActivityThread.main(ActivityThread.java:5624)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:959)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:754)
```

### PR DESCRIPTION
**Pull request by tonyr59h**
Fixes #5971. Not sure exactly why the selected sites list is out of whack but this will at least address the crash. I included a similar check in a sister method since we can't guarantee the validity of the selected site list.

To test:
* I just added a random ID to the list and went into action mode in the site picker (long press .com site)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java [hunk 1]
```diff
@@ -267,7 +267,7 @@ int getNumSelected() {
     int getNumHiddenSelected() {
         int numHidden = 0;
         for (Integer i: mSelectedPositions) {
-            if (mSites.get(i).isHidden) {
+            if (i < mSites.size() && mSites.get(i).isHidden) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java]: Can we use `isValidPosition()` here instead?

Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java]: Changed in 3266cf52de6c7e8c312e0e2859fa2e0ebf7058ba

**Comment by nbradbury** [comment]
@tonyr59h Looks good, but it's failing Travis now for some reason.

**Comment by aforcier** [comment]
> @tonyr59h Looks good, but it's failing Travis now for some reason.

This happened to me as well - the problem is that the editor lib was using a dynamic dependency for the wp utils library, which had its support lib versions [updated](https://github.com/wordpress-mobile/WordPress-Utils-Android/commit/a2efde8e8890fff29a61fa1693b893ec21b8a259). This branch doesn't have the changes to WPAndroid `develop` that update to the same support lib versions (https://github.com/wordpress-mobile/WordPress-Android/pull/5994), and that's causing a support library version conflict.

Merging `develop` in here should fix Travis, and I've [switched](https://github.com/wordpress-mobile/WordPress-Android/commit/832c42e42ca6cf4c64b303ea9a0fe565552cff55) the editor lib back to static versions for utils so this doesn't happen anymore.

**Comment by tonyr59h** [comment]
Thanks for the info @aforcier! Merged up 🤞

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**